protocol.go

```diff
@@ -0,0 +1,44 @@
+package tarantool
+
+type ProtocolVersion uint64
+type Feature uint64
+
+type protocolInfo struct {
+	version  ProtocolVersion
+	features []Feature
+}
+
+const ProtocolVersionUnsupported ProtocolVersion = 0
+
+const (
+	// Streams support.
+	FeatureStreams Feature = 0
+	// Interactive transactions support.
+	FeatureTransactions Feature = 1
+	// Support of MP_ERROR object over MessagePack.
+	FeatureErrorExtension Feature = 2
+	// Support of watchers.
+	FeatureWatchers Feature = 3
+)
```

Comment:
```suggestion
const (
	// Streams support.
	StreamsFeature Feature = 0
	// Interactive transactions support.
	TransactionsFeature Feature = 1
	// Support of MP_ERROR object over MessagePack.
	ErrorExtensionFeature Feature = 2
	// Support of watchers.
	WatchersFeature Feature = 3
)
```

Let's unify it here:

https://github.com/tarantool/go-tarantool/blob/master/stream.go#L9-L23
https://github.com/tarantool/go-tarantool/blob/master/datetime/adjust.go#L5-L11
https://github.com/tarantool/go-tarantool/blob/master/connection_pool/state.go#L8-L14